tests/Integration/Behaviour/Features/Context/Domain/CartFeatureContext.php

```diff
@@ -98,22 +99,24 @@ public function createEmptyCartForCustomer(string $cartReference, string $custom
      */
     public function addProductsToCarts(int $quantity, string $productName, string $cartReference)
     {
-        /** @var array $productsMap */
-        $productsMap = $this->getQueryBus()->handle(new SearchProducts($productName, 1, Context::getContext()->currency->iso_code));
-        $productId = array_key_first($productsMap);
+        /** @var array $products */
+        $products = $this->getQueryBus()->handle(new SearchProducts($productName, 1, Context::getContext()->currency->iso_code));
 
-        if (!$productId) {
-            throw new RuntimeException(sprintf('Product with name "%s" does not exist', $productName));
+        if (empty($products)) {
+            throw new RuntimeException(sprintf('Product with name "%s" was not found', $productName));
         }
 
+        /** @var FoundProduct $product */
+        $product = $products[0];
```

Comment:
```suggestion
        $product = reset($products);
```

Reply:
didn't know about this method :O looks better indeed, thanks :smile_cat: